File: ResourceManager/ViewModels/PageContentViewModel.vb
Imports VBNetCore.Mvvm.Abstractions
Imports VBNetCore.Mvvm.Commands
Imports System.Windows.Input
Imports System.Windows.Media
Imports System.Windows.Media.Imaging

Namespace ResourceManager.ViewModels
    Public Class PageContentViewModel
        Inherits ViewModelBase
        Private _MenuTitle As String, _MenuIcon As BitmapImage, _Title As String
        Public Sub New()
            SelectedMenuCommand = New RelayCommand(Sub() CType(mainVM.CurrentPage, PageViewModel).SelectedItem = Me)
        End Sub
        Public ReadOnly Property SelectedMenuCommand As ICommand

        Public Property MenuTitle As String
            Get
                Return _MenuTitle
            End Get
            Protected Set(ByVal value As String)
                _MenuTitle = value
            End Set
        End Property

        Public Property MenuIcon As BitmapImage
            Get
                Return _MenuIcon
            End Get
            Protected Set(ByVal value As BitmapImage)
                _MenuIcon = value
            End Set
        End Property

        Public Property Title As String
            Get
                Return _Title
            End Get
            Protected Set(ByVal value As String)
                _Title = value
            End Set
        End Property
        Public Property PageColor As Brush
    End Class
End Namespace

File: ResourceManager/ViewModels/PageContentViewModel.cs
using NetCore.Mvvm.Abstractions;
using NetCore.Mvvm.Commands;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Imaging;

namespace ResourceManager.ViewModels
{
    public class PageContentViewModel : ViewModelBase
    {
        public PageContentViewModel() 
        {
            SelectedMenuCommand = new RelayCommand(() => ((PageViewModel)mainVM.CurrentPage).SelectedItem = this);  
        }
        public ICommand SelectedMenuCommand { get; }
        public string MenuTitle { get; protected set; }
        public BitmapImage MenuIcon { get; protected set; }
        public string Title { get; protected set; }
        public Brush PageColor { get; set; }
    }
}
